presto-main/src/main/java/com/facebook/presto/operator/scalar/ArrayLessThanOperator.java

```diff
@@ -49,18 +54,45 @@ private ArrayLessThanOperator()
     @Override
     public FunctionInfo specialize(Map<String, Type> types, int arity, TypeManager typeManager, FunctionRegistry functionRegistry)
     {
-        Type type = types.get("T");
-        type = typeManager.getParameterizedType(StandardTypes.ARRAY, ImmutableList.of(type.getTypeSignature()), ImmutableList.of());
+        Type elementType = types.get("T");
+        Type type = typeManager.getParameterizedType(StandardTypes.ARRAY, ImmutableList.of(elementType.getTypeSignature()), ImmutableList.of());
         TypeSignature typeSignature = type.getTypeSignature();
-        return operatorInfo(LESS_THAN, RETURN_TYPE, ImmutableList.of(typeSignature, typeSignature), METHOD_HANDLE.bindTo(type), false, ImmutableList.of(false, false));
+        MethodHandle lessThanFunction = functionRegistry.resolveOperator(LESS_THAN, ImmutableList.of(elementType, elementType)).getMethodHandle();
+        MethodHandle methodHandle = methodHandle(ArrayLessThanOperator.class, "lessThan", MethodHandle.class, Type.class, Slice.class, Slice.class);
+        MethodHandle method = methodHandle.bindTo(lessThanFunction).bindTo(elementType);
+        return operatorInfo(LESS_THAN, RETURN_TYPE, ImmutableList.of(typeSignature, typeSignature), method, false, ImmutableList.of(false, false));
     }
 
-    public static boolean lessThan(Type type, Slice left, Slice right)
+    public static boolean lessThan(MethodHandle lessThanFunction, Type type, Slice left, Slice right)
     {
-        BlockBuilder leftBlockBuilder = type.createBlockBuilder(new BlockBuilderStatus(), 1, left.length());
-        BlockBuilder rightBlockBuilder = type.createBlockBuilder(new BlockBuilderStatus(), 1, right.length());
-        leftBlockBuilder.writeBytes(left, 0, left.length());
-        rightBlockBuilder.writeBytes(right, 0, right.length());
-        return type.compareTo(leftBlockBuilder.closeEntry().build(), 0, rightBlockBuilder.closeEntry().build(), 0) < 0;
+        Block leftArray = readStructuralBlock(left);
+        Block rightArray = readStructuralBlock(right);
+
+        int len = Math.min(leftArray.getPositionCount(), rightArray.getPositionCount());
+        int index = 0;
+        while (index < len) {
+            checkElementNotNull(leftArray.isNull(index), ARRAY_NULL_ELEMENT_MSG);
+            checkElementNotNull(rightArray.isNull(index), ARRAY_NULL_ELEMENT_MSG);
+            Object leftElement = castValue(type, leftArray, index);
+            Object rightElement = castValue(type, rightArray, index);
+            try {
+                if (((Boolean) lessThanFunction.invoke(leftElement, rightElement)).booleanValue()) {
+                    return true;
+                }
```

Comment:
I think we should optimize this so that if leftElement > rightElement it returns false. You can just get the greaterThanFunction handle too